data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/extension/MaxMindConfigTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ *  PDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.extension;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+import java.time.Duration;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class MaxMindConfigTest {
+    private MaxMindConfig maxMindConfig;
+
+    @BeforeEach
+    void setup() {
+        maxMindConfig = new MaxMindConfig();
+    }
+
+    @Test
+    void testDefaultConfig() {
+        assertThat(maxMindConfig.getDatabasePaths().size(), equalTo(0));
+        assertThat(maxMindConfig.getDatabaseRefreshInterval(), equalTo(Duration.ofDays(7)));
+        assertThat(maxMindConfig.getCacheSize(), equalTo(4096));
+    }
+
+    @Test
+    void testCustomConfig() throws NoSuchFieldException, IllegalAccessException {
+        reflectivelySetField(maxMindConfig, "databaseRefreshInterval", Duration.ofDays(10));
+        reflectivelySetField(maxMindConfig, "cacheSize", 2048);
+        reflectivelySetField(maxMindConfig, "databasePaths", List.of("path1", "path2", "path3"));
+
+        assertThat(maxMindConfig.getDatabaseRefreshInterval(), equalTo(Duration.ofDays(10)));
+        assertThat(maxMindConfig.getCacheSize(), equalTo(2048));
+        assertThat(maxMindConfig.getDatabasePaths().size(), equalTo(3));
+    }
+
+    private void reflectivelySetField(final MaxMindConfig maxMindConfig, final String fieldName, final Object value) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Let's use the common implementation.

https://github.com/opensearch-project/data-prepper/blob/6648d9fea792cfca889c69f62ce50af6ee12392a/data-prepper-test-common/src/main/java/org/opensearch/dataprepper/test/helper/ReflectivelySetField.java#L29